cpp/include/raft/random/rng.cuh

```diff
@@ -722,6 +722,8 @@ template <typename T>
 using weight_t = typename weight_alias<T>::type;
 }  // namespace sample_without_replacement_impl
 
+//@{
```

Comment:
Should we add something like `@defgroup sample without replacement calls` here to indicate what this group is?

Reply:
I'd like to move this out into its own file in 23.02. Sphinx seems to be having trouble resolving it and I'm not sure if it's related to sphinx, breathe, or doxygen. I'll create an issue for it.

Reply:
Thanks for the suggestion here! Using `defgroup` did fix the ambiguity issues I was facing w/ the `sample_with_replacement` function so I went ahead and added that here. I'm thinking we should use defgroup more throughout the codebase in general to isolate and document only the `mdspan`-based APIs everywhere.

Reply:
I'll leave the issue open though, since the other piece is moving `sample_without_replace` into its own file.